packages/codemods/src/codeMods/ICodeMod.ts

```diff
@@ -0,0 +1,28 @@
+import { SourceFile } from 'ts-morph';
+
+export interface ICodeModResult {
+  success?: boolean;
+}
+
+export interface ICodeMod<T = SourceFile> {
```

Comment:
Depending on how T is used (haven't finished reading yet) you may need to put an `extends` constraint on it instead of just supplying a default. Also what other types would you expect it to potentially be?

Reply:
This is more of an escape hatch right now. We're expecting it to basically always be of type SourceFile, but if you wanted to test, say a codemod which just takes a string, you can do that.